core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -279,6 +282,10 @@ class ContainerProxy(factory: (TransactionId,
   when(Uninitialized) {
     // pre warm a container (creates a stem cell container)
     case Event(job: Start, _) =>
+      val deadline = job.ttl match {
+        case Some(value) => Some(FiniteDuration(value.toMillis, TimeUnit.MILLISECONDS).fromNow)
```

Comment:
with above change from Duration to FiniteDuration, you can now just `case Some(value) => Some(value.fromNow)`, or better just remove the match and below you can 
`PreWarmCompleted(PreWarmedData(container, job.exec.kind, job.memoryLimit, expires = job.ttl.map(_.fromNow))))`

Reply:
Updated accordingly